test/widgets/action_sheet_test.dart

```diff
@@ -99,46 +107,244 @@ void main() {
     connection.prepare(httpStatus: 400, json: fakeResponseJson);
   }
 
-  group('AddThumbsUpButton', () {
-    Future<void> tapButton(WidgetTester tester) async {
-      await tester.ensureVisible(find.byIcon(ZulipIcons.smile, skipOffstage: false));
-      await tester.tap(find.byIcon(ZulipIcons.smile));
-      await tester.pump(); // [MenuItemButton.onPressed] called in a post-frame callback: flutter/flutter@e4a39fa2e
-    }
+  group('ReactionButtons', () {
+    final popularCandidates = EmojiStore.popularEmojiCandidates;
 
-    testWidgets('success', (tester) async {
-      final message = eg.streamMessage();
-      await setupToMessageActionSheet(tester, message: message, narrow: TopicNarrow.ofMessage(message));
+    group('popular emoji reactions;', () {
+      testWidgets('ensure all are shown', (tester) async {
+        final message = eg.streamMessage();
+        await setupToMessageActionSheet(tester, message: message, narrow: TopicNarrow.ofMessage(message));
 
-      connection.prepare(json: {});
-      await tapButton(tester);
-      await tester.pump(Duration.zero);
+        // Ensure all popular emoji buttons are shown.
+        final emojis = tester.widgetList<UnicodeEmojiWidget>(find.descendant(
+          of: find.descendant(
+            of: find.byType(ReactionButtons),
+            matching: find.byType(InkWell)),
+          matching: find.byType(UnicodeEmojiWidget)));
+        check(emojis).deepEquals(popularCandidates.map<Condition<Object?>>((emoji) {
+          final emojiDisplay = emoji.emojiDisplay as UnicodeEmojiDisplay;
+          return (it) => it.isA<UnicodeEmojiWidget>()
+            ..emojiDisplay.which((it) => it
+              ..emojiName.equals(emojiDisplay.emojiName)
+              ..emojiUnicode.equals(emojiDisplay.emojiUnicode));
+        }));
+      });
 
-      check(connection.lastRequest).isA<http.Request>()
-        ..method.equals('POST')
-        ..url.path.equals('/api/v1/messages/${message.id}/reactions')
-        ..bodyFields.deepEquals({
-            'reaction_type': 'unicode_emoji',
-            'emoji_code': '1f44d',
-            'emoji_name': '+1',
+      for (final emoji in popularCandidates) {
+        final emojiDisplay = emoji.emojiDisplay as UnicodeEmojiDisplay;
+
+        Future<void> tapButton(WidgetTester tester) async {
+          await tester.tap(find.descendant(
+            of: find.descendant(
+              of: find.descendant(
+                of: find.byType(ReactionButtons),
+                matching: find.byType(InkWell)),
+              matching: find.byType(UnicodeEmojiWidget)),
+            matching: find.text(emojiDisplay.emojiUnicode)));
```

Comment:
We can put this more in terms of how the user sees it, and simplify:
```suggestion
        Future<void> tapButton(WidgetTester tester) async {
          await tester.tap(find.descendant(
            of: find.byType(BottomSheet),
            matching: find.text(emojiDisplay.emojiUnicode)));
```

(I see just `find.text` alone doesn't work in the "removing" tests, because the same emoji appears in a reaction chip.)